src/rewriter.coffee

```diff
@@ -539,7 +539,7 @@ exports.Rewriter = class Rewriter
   fixOutdentLocationData: ->
     @scanTokens (token, i, tokens) ->
       return 1 unless token[0] is 'OUTDENT' or
-        (token.generated and token[0] is 'CALL_END') or
+        (token.generated and token[0] is 'CALL_END' and not token.data?.closingTagNameToken) or
```

Comment:
After correctly wiring up the `generated` option in `Lexer::makeToken()`, location data for JSX elements was breaking since the rewriter wanted to reassign "real" location data to the generated JSX pseudo-tokens

So need to tell it to ignore this particular type of JSX pseudo-token and respect its (pseudo) location data

Basically this is patching a seam related to the fact that we don't really parse JSX and instead pass through tokens that don't actually exist